schema/gen/go/testEngine_disabled_test.go

```diff
@@ -33,14 +37,15 @@ func buildGennedCode(t *testing.T, prefix string, pkgName string) {
 	cmd := exec.Command("go", args...)
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
-	err := cmd.Run()
+	err = cmd.Run()
 	if err != nil {
 		t.Fatalf("genned code failed to compile: %s", err)
 	}
+
+	t.Skip("behavioral tests for generated code skipped: you used the 'skipgenbehavtests' build tag.")
```

Comment:
I think this probably should still be wrapped in some t.Run clause?  Because _some_ things are still appropriate to flag as passing if we get this far.

Reply:
Ah, will do, thanks.

Reply:
Actually, I can't do this, because there's no longer a `runBehavioralTests` layer with a sub-test. There's just engine Init + PrototypeByName.

I could go to every `func SchemaTest...` and wrap all the "test the prototypes" logic with a t.Run and asking the engine if it just wants to skip, but that's too much repeated boilerplate.

I could perhaps go back to a double-callback system where the engine is in charge of running all the behavioral tests, so it can run them in a sub-test and however it wants, but then the interface gets kinda complex and confusing.

I get that marking "up to this point" as passing can be useful, but I don't think it's to the point that we should turn the interface around that way. I'd rather have us move towards just always running the proper tests no matter the build tags, which would actually be possible now that the "behavioral test" logic is in libraries. For example:

* Run the code generator outputting to /tmp/somedir
* Generate an extra file with an Engine implementation, using the same PrototypeByName we generate today
* Have it call the right node/tests.SchemaTest func with the Engine (probably via a replace directive in a temporary module)
* Build a binary and run it, verifying it succeeds

So that would be akin to a plugin, but with a normal binary that either succeeds or errors in some way. About the same speed, and should work everywhere no matter cgo.

Reply:
Pretty sure this would let us get rid of the boilerplate in schema/gen/go, too. Happy to briefly chat about it later, and we could take a crack at it.